### ISSUE TITLE
Crash on Multiplayer

### ISSUE THREAD

**Issue by Jorgee9001** [OE100/ReHooked#1]
Minecraft version 1.20.1

Forge 47.3.0

Using the Iron hook crashes people on the server, other than the person using the hook. It's not always immediate, but it always happens eventually. Not sure whether its that hook in particular or all hooks.

[crash-2024-09-15_17.20.42-client.txt](https://github.com/user-attachments/files/17007674/crash-2024-09-15_17.20.42-client.txt)

**Comment by ghost** [comment]
I think I see the problem and can at least push out a temporary fix.
I'm sorry for the issues it caused you, the network code is still very much w.i.p as I am not yet experienced with it, I think it might even need a complete rewrite once I finish my assignments and get some more free time.

tldr: should be fixed in 1.2.1 which should be out in at most a couple hours once I finish coding and curseforge approves it.

**Comment by ghost** [comment]
@Jorgee9001 on a second though I'll keep the issue open, if you get the chance try playing with version 1.2.1, it should fix this issue, just uploaded it to github, curseforge is still processing.
I'd be happy to know if it works for you too.
Once again sorry for the inconvenience and have fun!
